### PR TITLE
Prioritise lambda response and if there is a failure by either, error…

### PR DESCRIPTION
**Pull request by markrichardsg**
… in the hope that the app re-sends validation

### CODE REVIEW

File: iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala [hunk 1]
```diff
@@ -64,25 +64,26 @@ class DelegatingValidateReceiptCompators(cloudWatch: CloudWatch) extends Delegat
...
+
           }
         }
 
       }
       case (Some(validateResponse), _) => {
         logger.warn(s"Validate mismatch for Request: $lambdaRequest \nLambda Response: $lambdaResponse \nDelegate Response: $delegateResponse")
         logDelegateOnly(validateResponse)
-        lambdaResponse
+        delegateResponse
```
Comment (alexduf) [inline iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala]: You're returning None here, why?

File: iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala [hunk 2]
```diff
@@ -64,25 +64,26 @@ class DelegatingValidateReceiptCompators(cloudWatch: CloudWatch) extends Delegat
...
       }
       case (_, Some(validateResponse)) => {
         logger.warn(s"Validate mismatch for Request: $lambdaRequest \nLambda Response: $lambdaResponse \nDelegate Response: $delegateResponse")
         logLambdaOnly(validateResponse)
-        delegateResponse
+        lambdaResponse
```
Comment (alexduf) [inline iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala]: Same here

File: iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala (hunk 1 above)
Comment (markrichardsg) [reply iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala]: If the validation succeeded in lambda but not delegate or vice versa, then there will be inconsistency in the purchases stored (eg: app store timeout)
If the app receives a bad response then the transactions should be re-sent by the device in the future and hopefully fix this inconsitency.

Comment (alexduf) [reply iosvalidatereceipts/src/main/scala/com/gu/mobilepurchases/lambda/DelegatingValidateReceiptLambda.scala]: Got it

**Pull request merged**